locator/guice/src/main/java/org/eclipse/kapua/locator/guice/KapuaLocatorImpl.java

```diff
@@ -117,7 +117,10 @@
         for (Binding<?> binding : bindings.values()) {
             final Class<?> clazz = binding.getKey().getTypeLiteral().getRawType();
             if (KapuaService.class.isAssignableFrom(clazz)) {
-                servicesList.add(injector.getInstance(clazz.asSubclass(KapuaService.class)));
+                KapuaService serviceInstance = injector.getInstance(clazz.asSubclass(KapuaService.class));
```

Comment:
I recommended Set as an option, but it would change method signature.
Also I am not sure if order of items is important and how are lists iterated over.